### ISSUE TITLE
Prow is not correctly managing the needs-triage labels

### ISSUE THREAD

**Issue by codificat** [thoth-station/thoth-application#2359]
**Describe the bug**

On some (most?) of the repos that are managed by prow, when adding a label about a triage status to an issue, by adding a comment with a command like `/triage foo`, the new label gets added (`triage/foo`). That works. But if the issue had a pre-existing label `needs-triage`, this one is not removed.

**To Reproduce**

Some examples where this happened:

 - https://github.com/thoth-station/search/issues/54#event-5990795463
 - https://github.com/thoth-station/search/issues/52#event-5990800381
 - https://github.com/thoth-station/adviser/issues/2245#issuecomment-1014417257
 - https://github.com/thoth-station/helm-charts/issues/29#event-6073632086

On other repos, like `core`, it behaves as expected when the `triage/accepted` label is added:

- https://github.com/thoth-station/core/issues/345#event-6075939402

but not with other `triage/` labels:

- https://github.com/thoth-station/core/issues/366#event-6075592842

Furthermore, the `needs-triage` label can not be removed:

- https://github.com/thoth-station/core/issues/366#issuecomment-1040491379

**Expected behavior**

Issues that have a `triage/foo` label –presumably added manually by a human– should not have a `needs-triage` label as well, and the bot should take care of avoiding this situation.

**Additional context**

There is a view in triage party that shows issues in this situation: see "needs-triage but has triage" in the [inconsistencies tab](https://triage-party.thoth-station.ninja/s/inconsistent?player=0&players=1).

The `require_matching_label` is [currently configured](https://github.com/thoth-station/thoth-application/blob/a23fde17dd03cc1998accccc96f827bdcdba638c/prow/overlays/smaug/plugins.yaml#L45) in a few repos for the needs-triage label.

**Comment by codificat** [comment]
Proposed path to resolution:

- [ ] remove the `needs-triage` label by default across [all repos](https://github.com/thoth-station/core/blob/00e7390b2c4af55578e736ac7b2dd56312851862/community/labels.yaml#L429-L434), leaving it only available on a few specific repos: the ones that are more "externally facing" like: core, thoth-application, support.
- [ ] ensure the `requrire_matching_label` prow plugin manages the `needs-triage` label on [these few repos](https://github.com/thoth-station/thoth-application/blob/a23fde17dd03cc1998accccc96f827bdcdba638c/prow/overlays/smaug/plugins.yaml#L45) (only)
- [ ] remove the `needs-triage` label from all GitHub issue templates, on all repos (the prow plugin will add it when/if needed) (https://github.com/thoth-station/template-project/pull/33)
- [ ] do not rely on the `needs-triage` label for triaging. Instead, look for the presence or lack of a `triage/*` label. Configure triage-party accordingly

Basically this means to treat the `needs-triage` label as a user experience feature to help inform users of the lifecycle of an issue, thanks to the friendly message posted by the prow plugin when it adds the label.
